clients/java/signalr/src/main/java/com/microsoft/aspnet/signalr/HubConnection.java

```diff
@@ -92,20 +100,23 @@ public HubConnection(String url, Transport transport, Logger logger){
             }
         };
 
-        if (transport == null){
-            try {
-                this.transport = new WebSocketTransport(this.url, this.logger);
-            } catch (URISyntaxException e) {
-                e.printStackTrace();
-            }
-        } else {
+        if (transport != null) {
             this.transport = transport;
         }
     }
 
+    public HubConnection(String url, Transport transport, Logger logger) {
```

Comment:
Why did you add 2 constructors?

Reply:
For testing. Once I add the skipNegotiate option to the builder I can clean this up